libs/pushstreams/CMakeLists.txt

```diff
@@ -16,10 +16,11 @@
 # under the License.
 
 set(PUSHSTREAMS_DEFAULT_ON ${CELIX_CXX17})
-celix_subproject(PUSHSTREAMS "Option to build the PushStreams library" ${PUSHSTREAMS_DEFAULT_ON} DEPS PROMISES)
+celix_subproject(PUSHSTREAMS "Option to build the PushStreams library" ${PUSHSTREAMS_DEFAULT_ON})
```

Comment:
Does this mean that the celix cmake function `celix_subproject` is no longer used to configure dependecies, but this is now done using the Conanfile?

So for pushstream the lines :
```
        if self.options.build_pushstreams:
            self.options.build_promises = True
```

Reply:
Yes. Previously `celix_subproject` will only hint our users to enable missing components without automatic configuration.
For a long chain of dependencies, users have to try several times to work out a working configuration for Celix.
Imagine that we have several Conan packages, each of them has intra-package dependencies like Celix, and they also have inter-package dependencies. Then working out a working configuration for a downstream user can be daunting.
With this in place, our users only specify what they need directly, then everything else is automatically configured.